opencl/source/sharings/gl/linux/gl_sharing_linux.cpp

```diff
@@ -11,41 +11,16 @@
 #include "opencl/source/helpers/gl_helper.h"
 #include "opencl/source/sharings/gl/gl_arb_sync_event.h"
 
-namespace Os {
-extern const char *eglDllName;
-extern const char *openglDllName;
-} // namespace Os
-
 namespace NEO {
 GLSharingFunctionsLinux::GLSharingFunctionsLinux(GLType glhdcType, GLContext glhglrcHandle, GLContext glhglrcHandleBkpCtx, GLDisplay glhdcHandle)
     : glHDCType(glhdcType), glHGLRCHandle(glhglrcHandle), glHGLRCHandleBkpCtx(glhglrcHandleBkpCtx), glHDCHandle(glhdcHandle) {
     GLSharingFunctionsLinux::initGLFunctions();
-    updateOpenGLContext();
 }
 GLSharingFunctionsLinux::~GLSharingFunctionsLinux() = default;
 
 bool GLSharingFunctionsLinux::isGlSharingEnabled() {
-    static bool oglLibAvailable = std::unique_ptr<OsLibrary>(OsLibrary::load(Os::eglDllName)).get() != nullptr;
-    return oglLibAvailable;
-}
-
-void GLSharingFunctionsLinux::createBackupContext() {
-    if (pfnEglCreateContext) {
-        glHGLRCHandleBkpCtx = pfnEglCreateContext(glHDCHandle);
-        pfnEglShareLists(glHGLRCHandle, glHGLRCHandleBkpCtx);
-    }
-}
-
-GLboolean GLSharingFunctionsLinux::setSharedOCLContextState() {
-    ContextInfo contextInfo{};
-    GLboolean retVal = glSetSharedOCLContextState(glHDCHandle, glHGLRCHandle, CL_TRUE, &contextInfo);
-    if (retVal == GL_FALSE) {
-        return GL_FALSE;
-    }
-    glContextHandle = contextInfo.contextHandle;
-    glDeviceHandle = contextInfo.deviceHandle;
-
-    return retVal;
+    std::unique_ptr<OsLibrary> dynLibrary(OsLibrary::load(""));;
```

Comment:
So there are several things :

(1) We can't link directly to any GL library. We have to use which ever one the application is using and it might be a special one for whatever reason, so we can't add `-lGL` to the Makefile or load any library by name explicitely.

(2) We can't just "use" GL symbol without linking and assume they will be present because if we do, then any application that doesn't use GL interop and doesn't use GL at all will end up with a "symbol not found" error.

(3) But we know that _if_ the user want to use GL interop, then at somepoint, it will have to have loaded a GL library in the global scope which mean at that point we can query it to find the few functions we need. And if that lookup fails, we know the user will not need GL interop.

Reply:
(1) - I know that we cannot link using load-time linking as GL sharing is optional but what is an issue with run-time linking by name? When user is using GL interop then GL library need to be available in the system
(2) I totally agree, we cannot use GL symbols directly 
(3) This approach assumes that GL is loaded before OCL is loaded but I can imagine a scenario where CL is initialized before GL is even loaded and in such case I hope should try to load GL to determine if sharing is available.